packages/infrastructure/mongodb_atlas_identity_provider/README.md

```diff
@@ -0,0 +1,73 @@
+# Mongodb Atlas MongoDB Identity Provider
+
+This module sets up the identity provider configuration in MongoDB Atlas.
+
+## Guide
+
+This guide will help setup the MongoDB Atlas side of the SSO integration.
+
+### Pre-req: Setup MongoDB Atlas Identity Provider Domain Verification
+
+1. Go to your MongoDB Atlas account
+2. Go to `Organization Settings` -> `Federated Authentication Settings` -> `Domains`
+3. Click on `Add Domain`
+4. Add the root domain that you are using for Authentik (ie: panfactum.com)
+5. Select `DNS Record` as the verification method
+6. Note the `TXT Record` that is generated
+
+### Deploy a new DNS TXT Record & Verify
+
+1. Add a text record to your `global/aws_dns/terragrunt.hcl` file that looks like [this](https://github.com/Panfactum/stack/blob/__PANFACTUM_VERSION_EDGE__/packages/reference/environments/production/global/aws_dns_records/terragrunt.hcl#118).
+2. Run `terragrunt apply`
+3. Go back to MongoDB Atlas and click on `Verify` next to the domain you added
+4. Confirm verification
+
+### Download the Signing Certificate from Authentik
+
+Before proceeding, make sure to have downloaded the signing certificate from the Authentik application
+1. Login to your Authentik instance
+2. Switch over to the `Admin` section
+3. Go to the section `Applications` -> `Providers` -> `MongoDB Atlas`
+4. Find `Related objects` and click on the `Download` button for the `Download signing certificate`
+
+### Deploy the MongoDB Identity Provider module
+
+Unfortunately, the terraform provider for MongoDB Atlas does not support the creation of the Identity Provider but allows for modifications. 
+We will first create the resource through the UI and then import it to configure further.
+
+From MongoDB Atlas UI
+1. Go to `Organization Settings` -> `Federated Authentication Settings` -> `Identity Providers`
```

Comment:
I am fairly confident that none of this has to be done manually if you use the OIDC provider instead of the SAML provider. With the OIDC provider, you do not need a certificate copy step and the rest can be filled programmatically.

Reply:
@fullykubed the UI selections and description in atlas suggest that SAML is to be used to provide access to their web admin vs OIDC if for accessing the db. My knowledge is limitted here but do you think this is just a suggestion and OIDC can be used? 

![image](https://github.com/user-attachments/assets/d375145b-6e8a-4663-8858-8723a2a2b654)

Reply:
This also suggest that it must be SAML. 
`To configure Federated Authentication, you must have an external SAML IdP application. In the SAML IdP, you must perform the following:`

https://www.mongodb.com/docs/atlas/security/manage-federated-auth/#configure-an-external-identity-provider-application

Reply:
Hmmm... ok. 

I guess I am not entirely sure why there would be two systems? Are we hooking up both in this module?

I think we'd want to ensure SSO is setup for both the UI and the databases, right?

Reply:
Whichever direction you choose, can you add some detailed documentation about _why_ you made the relevant design decisions? At this point, you know more than I do about mongodb atlas, and I am just going off of the fields I see in the tf provider.